core/utils/backoff_ticker.go

```diff
@@ -0,0 +1,85 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jpillora/backoff"
+)
+
+type BackoffTicker struct {
+	b         backoff.Backoff
+	timer     *time.Timer
+	C         chan time.Time
+	chStop    chan struct{}
+	isRunning bool
+	sync.RWMutex
+}
+
+func NewBackoffTicker(min, max time.Duration) BackoffTicker {
+	c := make(chan time.Time, 1)
```

Comment:
My gut tells me that with a primitive such as this, which we want to make as reusable as possible across the codebase, it'd be best to be extremely unopinionated about the channel concurrency and force consumers of this type to decide whether they want to block, discard if not ready, or queue up an extra run if the consumer is already busy. Concrete suggestion: make it unbuffered and move the buffering into FM. If it's helpful, the relationship between synchronous LogBroadcaster delivery and the FM BoundedPriorityQueue is the same as what I'm thinking here

Reply:
I intentionally implemented the BackoffTicker in the similar style as a `time.Ticker` and `utils.PausableTicker`. While 'm open to changing it, I do like that you can keep the same mental model as dealing with any other ticker